src/canmatrix/tests/test_canmatrix.py

```diff
@@ -145,6 +147,138 @@ def test_ecu_repr():
     assert str(ecu) == "BoardUnit(name='Gateway', comment='with bug')"
 
 
+# Signal (generic functions)
+def test_signal_has_comment(some_signal):
+    comment = "comment01"
+    some_signal.addComment(comment)
+    assert some_signal.comment == comment
+
+
+def test_signal_find_attribute(some_signal):
+    some_signal.addAttribute("attr1", 255)
+    assert some_signal.attribute("attr1") == 255  # check value from 'attributes'
```

Comment:
Do these comments explain something the code doesn't?  Ditto elsewhere.

Reply:
I think that looking on test code it isn't obvious why testing "attr1" and "is_float" and also another values in the next TC. I wanted to state that one value is from class attribute, another from "user defined" `attributes` dict et so on. Maybe not well formulated.

Reply:
Yeah, I didn't get that, sorry.  I'd say that's two tests.  One to read 'builtin' attributes or whatever, and another for 'custom' or 'user'.  Whatever terminology is used elsewhere (if it's clear).  Then it moves what you wanted to express via the comments into code (and even test results).

Reply:
test splitted